packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -487,7 +487,10 @@ export class ElectronMainApplication {
 
     protected async handleStopRequest(electronWindow: BrowserWindow, onSafeCallback: () => unknown, reason: StopReason): Promise<void> {
         // Only confirm close to windows that have loaded our front end.
-        const safeToClose = !electronWindow.webContents.getURL().includes(this.globals.THEIA_FRONTEND_HTML_PATH) || await this.checkSafeToStop(electronWindow, reason);
+        // Since our resolved frontend HTML path might contain backward slashes (Windows), we normalize everything first.
+        const currentUrl = electronWindow.webContents.getURL().replace(/\\/g, '/');
+        const frontendUri = this.globals.THEIA_FRONTEND_HTML_PATH.replace(/\\/g, '/');
+        const safeToClose = !currentUrl.includes(frontendUri) || await this.checkSafeToStop(electronWindow, reason);
```

Comment:
Shouldn't we only replace the backslashes when on Windows?

Reply:
Sure, but this code doesn't do anything on any non-Windows-OS anyway (since no backwards slashes appear). But I'll wrap it in a `isWindows` branch.

Reply:
It's weird but IIRC unix systems allow backslashes in filenames:

```
$ touch \\meh\\
$ ls
'\meh\'
$ node -p 'fs.readdirSync(".").forEach(console.log)'
\meh\ 0 [ '\\meh\\' ]
undefined 
```

We should keep this in mind when converting paths to forward slashes and/or back.